src/extension/background-script/connectors/lnd.ts

```diff
@@ -223,11 +227,82 @@ class Lnd implements Connector {
     });
   };
 
-  getTransactions = () => {
-    return this.request("GET", "/v1/payments", undefined, {
-      transactions: [],
-    });
-  };
+  async getInvoices(): Promise<GetInvoicesResponse> {
+    const data = await this.request<{
+      invoices: {
+        add_index: string;
+        amt_paid_msat: string;
+        amt_paid_sat: string;
+        amt_paid: string;
+        cltv_expiry: string;
+        creation_date: string;
+        description_hash?: string;
+        expiry: string;
+        fallback_addr: string;
+        features: unknown[];
+        htlcs: {
+          chan_id: string;
+          htlc_index: string;
+          amt_msat: string;
+          accept_height: number;
+          accept_time: string;
+          resolve_time: string;
+          expiry_height: number;
+          state: "SETTLED";
+          custom_records: {
+            "696969": string;
+            "7629169": string;
+            "5482373484": string;
+          };
+          mpp_total_amt_msat: string;
+          amp?: unknown;
+        }[];
+        is_keysend: boolean;
+        memo: string;
+        payment_addr: string;
+        payment_request: string;
+        private: boolean;
+        r_hash: string;
+        r_preimage: string;
+        route_hints: [];
+        settle_date: string;
+        settle_index: string;
+        settled: boolean;
+        state: string;
+        value_msat: string;
+        value: string;
+      }[];
+      last_index_offset: string;
+      first_index_offset: string;
+    }>("GET", "/v1/invoices", undefined, {});
+
+    const invoices: Invoice[] = data.invoices
+      .map((invoice, index): Invoice => {
+        const custom_records =
+          invoice.htlcs[0] && invoice.htlcs[0].custom_records;
+
+        const boostagram = utils.getBoostagramFromInvoice(custom_records);
```

Comment:
wondering if this is the right place to do this, or if the connector should return the raw data and then on a "higher level" we make sense of the data / extract the boostagram. 

I'd move that to the action level actually.

Reply:
Moved this to the action